src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -80,7 +82,7 @@ public function loadUserByUsername($username)
             throw new UsernameNotFoundException('More than one user found');
         }
 
-        return $this->loadUser($username, $entries[0]);
+        return $this->loadUser($entries[0]->getAttribute($this->uidKey)[0], $entries[0]);
```

Comment:
Also, you should check whether the entry has the requested attribute (we need to provide an error message in case of misconfiguration of the LDAP attribute, or if the attribute is not returned by the server).